### DISCUSSION TITLE
My database function always returns nothing.

### DISCUSSION THREAD

**Discussion by fupicat** [supabase/supabase#8936] [Questions] [answered]
I'm new to databases so maybe this is more of a Postgres question.

I'm making a social media-like app where users (PROFILES) can submit little games (PROJECTS). Users can like projects, expressed by a table (LIKES) that relates a profile to which project they liked. Each project also has a like_count column, which I want to be updated every time a user likes or unlikes a project, so for that I'm using a database function. It returns a boolean: true if the project was successfully liked, and false if the user has already liked that project. I'll make a different function to handle unliking.

Here's my code:

```sql
create or replace function like_project(my_user_id uuid, my_project_id bigint) returns boolean as $$
declare
  liked_row likes;
  my_like_count bigint;
begin
  -- check if user has already liked
  select * from likes
  into liked_row
  where user_id = like_project.my_user_id
  and project_id = like_project.my_project_id;

  -- if user has already liked, don't like again, just return false.
  if found then
    return false;
  end if;
  -- else, you can like the project.

  -- insert new like
  insert into likes(user_id, project_id)
  values (like_project.my_user_id, like_project.my_project_id);

  -- count likes
  select count(*) from likes
  into my_like_count
  where project_id = like_project.my_project_id;

  -- update count on project row
  update projects
  set like_count = my_like_count
  where id = like_project.my_project_id;

  -- success
  return true;
end;
$$ language plpgsql;

-- test
select like_project('dc4dea31-a8c4-4bbf-87d5-13a731de2d91', 10);
```

The function works, except for the return value. My test query always outputs "Success. No rows returned" instead of true or false. What do I do to get the behavior I want?

**Comment by fupicat** [comment]
Follow up question: if I keep my function like this, could anyone theoretically add or remove likes from any account if they know the anon key? Can I check the ID of the current user in the function somehow, or make this function only executable server-side via the admin key?

**Reply by GaryAustin1** [reply]
You should use auth.uid() in the function instead of passing it.  Then it is only valid for the signed in user.

**Comment by GaryAustin1** [comment, marked as answer]
As far as testing your function.  Try creating (which you should have already done).
Then run the select function() on its own.

**Reply by fupicat** [reply]
It did work this time when tested on its own for some reason. Thank you for that, and also for answering my question about auth!
